### ISSUE TITLE
Show/hide column based upon callback

### ISSUE THREAD

**Issue by vagita** [tabalinas/jsgrid#1310]
Hi,
I'm trying to show/hide a column based upon the click of a checkbox in my grid.
{  name: "checkBoxField", type: "checkbox", width: 20, editTemplate: function(value){
			var $editControl = jsGrid.fields.checkbox.prototype.editTemplate.call(this, 
                        value,args[3]);
			  // Attach listenser !
                $editControl.click(function(){
                    var checkBoxValue = $(this).context.checked;	 
		   //toggle column visiblity
                     showHideColumnFunction(checkBoxValue, args[3], jsGrid);
                });
                return $editControl;
		}
	}
In the toggle function I tried
function showHideColumnFunction(value, row, grid){
                 if(value){
			jsGrid.fieldOption("maxFixChargesAmount", "visible", true);
		}
}

But it renders the whole grid again and escapes the edit mode. I can not figure out how to keep the editing mode enables.
Thanks in advance.

**Comment by tomvanlier** [comment]
Can't you put a css-selector on it (eg : css:"mycheckboxfield") and then put a callback outside of jsgrid like

```
  $('.mycheckboxfield').click(function(){
            if($(this).prop("checked") == true){
                 $("#mygrid").jsGrid("fieldOption", "Control", "visible", true); 
            }
            else if($(this).prop("checked") == false){
                $("#mygrid").jsGrid("fieldOption", "Control", "visible", false); 
            }
        });
```

**Comment by vagita** [comment]
It results in the same behavior, the grid is again reset and its no longer in edit mode.

**Comment by tomvanlier** [comment]
You are right, I delved deeper for the solution I needed and ended up modifying the source code.

Put something like this in your _createEditButton (before e.stopPropagation() and before grid.editItem() is called);

`grid.fieldOption("myGridField", "visible", true);`

And the negation in _createUpdateButton and _createCancelEditButton (also before e.stopPropagation());

`grid.fieldOption("myGridField", "visible", false;`

The idea behind it is that you make the desired column(s) visibile (all sorts of callbacks) and then launch the edititem when it's visible. You hide it agaian when updating or cancelling.

It works with multiple columns and with multiple grids on one page, you can do a caonditional on the first field name e.g.

Hope that works
